paymentsheet/src/main/java/com/stripe/android/paymentsheet/verticalmode/PaymentMethodVerticalLayoutInteractor.kt

```diff
@@ -14,48 +20,105 @@ internal interface PaymentMethodVerticalLayoutInteractor {
     data class State(
         val supportedPaymentMethods: List<SupportedPaymentMethod>,
         val isProcessing: Boolean,
+        val selectedPaymentMethodIndex: Int,
     )
 
     sealed interface ViewAction {
         data object TransitionToManageSavedPaymentMethods : ViewAction
-        data class TransitionToForm(val selectedPaymentMethodCode: String) : ViewAction
+        data class PaymentMethodSelected(val selectedPaymentMethodCode: String) : ViewAction
     }
 }
 
 internal class DefaultPaymentMethodVerticalLayoutInteractor(
-    private val viewModel: BaseSheetViewModel,
+    paymentMethodMetadata: PaymentMethodMetadata,
+    processing: StateFlow<Boolean>,
+    selection: StateFlow<PaymentSelection?>,
+    private val formElementsForCode: (code: String) -> List<FormElement>,
+    private val transitionTo: (screen: PaymentSheetScreen) -> Unit,
+    private val onFormFieldValuesChanged: (formValues: FormFieldValues, selectedPaymentMethodCode: String) -> Unit,
+    private val manageScreenFactory: () -> PaymentSheetScreen,
+    private val formScreenFactory: (selectedPaymentMethodCode: String) -> PaymentSheetScreen,
 ) : PaymentMethodVerticalLayoutInteractor {
+    constructor(viewModel: BaseSheetViewModel) : this(
+        paymentMethodMetadata = requireNotNull(viewModel.paymentMethodMetadata.value),
```

Comment:
It can never change when in the context of this interactor. And it's always going to be non null. So modeling differently just seems wrong here.

It can change in base sheet view model because it goes through the loading phase, where this is only shown if loading is successful.

Reply:
got it! Thanks for the explanation